pkg/strategy/xmaker/splithedge.go

```diff
@@ -0,0 +1,278 @@
+func (h *SplitHedge) InitializeAndBind(sessions map[string]*bbgo.ExchangeSession, strategy *Strategy) error {
+	// a simple guard clause
+	if !h.Enabled {
+		return nil
+	}
+
+	for name, config := range h.HedgeMarkets {
+		hedgeMarket, err := initializeHedgeMarketFromConfig(config, sessions)
+		if err != nil {
+			return err
+		}
+
+		h.hedgeMarketInstances[name] = hedgeMarket
+
+		hedgeMarket.Position.StrategyInstanceID = strategy.InstanceID()
+
+		// handle position cover and close here
+		hedgeMarket.positionExposure.OnCover(strategy.positionExposure.Cover)
+
+		hedgeMarket.tradeCollector.OnTrade(func(
+			trade types.Trade, profit fixedpoint.Value, netProfit fixedpoint.Value,
+		) {
+			strategy.positionExposure.Close(trade.PositionDelta())
+
+			if order, ok := hedgeMarket.orderStore.Get(trade.OrderID); ok {
+				strategy.orderStore.Add(order)
+			}
+
+			strategy.tradeCollector.ProcessTrade(trade) // this triggers position update and profit updates
+		})
+
+	}
+
+	h.strategy = strategy
+	h.logger = strategy.logger.WithField("component", "split_hedge")
+	return nil
+}
+
+func (h *SplitHedge) hedgeWithProportionAlgo(
+	ctx context.Context,
+	uncoveredPosition fixedpoint.Value,
+) error {
+	if h.ProportionAlgo == nil || len(h.ProportionAlgo.ProportionMarkets) == 0 {
+		return fmt.Errorf("proportion algo requires proportion markets")
+	}
+
+	delta := uncoveredToDelta(uncoveredPosition)
+	side := deltaToSide(delta)
+	remainingQuantity := delta.Abs()
+
+	for _, proportionMarket := range h.ProportionAlgo.ProportionMarkets {
+		hedgeMarket, ok := h.hedgeMarketInstances[proportionMarket.Name]
+		if !ok {
+			h.logger.Warnf("[splitHedge] hedge market %s not found", proportionMarket.Name)
+			continue
+		}
+
+		canHedge, err := hedgeMarket.canHedge(ctx, uncoveredPosition)
+		if err != nil {
+			h.logger.WithError(err).Errorf("[splitHedge] hedge market checking canHedge failed")
+			continue
+		} else if !canHedge {
+			h.logger.Infof("[splitHedge] hedge market %s cannot hedge now", proportionMarket.Name)
+			continue
+		}
+
+		bid, ask := hedgeMarket.getQuotePrice()
+		price := sideTakerPrice(bid, ask, side)
+		proportionQuantity := proportionMarket.CalculateQuantity(remainingQuantity, price)
+		proportionQuantity = AdjustHedgeQuantityWithAvailableBalance(
+			hedgeMarket.session.GetAccount(), hedgeMarket.market, side, proportionQuantity, price,
+		)
+
+		proportionQuantity = hedgeMarket.market.TruncateQuantity(proportionQuantity)
+
+		if proportionQuantity.IsZero() {
+			h.logger.Infof("skip zero quantity")
+			continue
+		}
+
+		if hedgeMarket.market.IsDustQuantity(proportionQuantity, price) {
+			h.logger.Infof("skip dust quantity: %s @ price %f", proportionQuantity.String(), price.Float64())
+			continue
+		}
+
+		remainingQuantity = remainingQuantity.Sub(proportionQuantity)
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case hedgeMarket.positionDeltaC <- quantityToDelta(proportionQuantity, side):
```

Comment:
Sending to `hedgeMarket.positionDeltaC` may block.
Is it by design?

Reply:
yes, it's critical and can not be skipped